libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m

```diff
@@ -155,12 +155,12 @@ - (void)pgQuerySoup:(CDVInvokedUrlCommand *)command
         NSDictionary *querySpecDict = [argsDict nonNullObjectForKey:kQuerySpecArg];
         SFQuerySpec* querySpec = [[SFQuerySpec alloc] initWithDictionary:querySpecDict withSoupName:soupName];
         [self log:SFLogLevelDebug format:@"pgQuerySoup with name: %@, querySpec: %@", soupName, querySpecDict];
-        NSError* error;
+        NSError* error = nil;
         SFStoreCursor* cursor = [self runQuery:querySpec error:&error argsDict:argsDict];
-        if (cursor) {
-            if ([self isGlobal:argsDict]) {
+        if (cursor.cursorId) {
+            if ([self isGlobal:argsDict] && self.globalCursorCache) {
                 (self.globalCursorCache)[cursor.cursorId] = cursor;
-            } else {
+            } else if (self.userCursorCache) {
```

Comment:
When does the cursor cache end up being nil?
Should we check for it not being nil in the other places where we access it?